### PR TITLE
Fixing 01584_distributed_buffer_cannot_find_column with analyzer.

### PR DESCRIPTION
**Pull request by KochetovNicolai**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 3def01f1f6951d11a142f80ef64f3d315ba92a4d with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61550/3def01f1f6951d11a142f80ef64f3d315ba92a4d/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Storages/StorageValues.cpp [hunk 1]
```diff
@@ -32,6 +46,25 @@ Pipe StorageValues::read(
...
+    if (!prepared_pipe.empty())
+    {
+        auto dag = std::make_shared<ActionsDAG>(prepared_pipe.getHeader().getColumnsWithTypeAndName());
+        ActionsDAG::NodeRawConstPtrs outputs;
+        outputs.reserve(column_names.size());
+        for (const auto & name : column_names)
+            outputs.push_back(dag->getOutputs()[prepared_pipe.getHeader().getPositionByName(name)]);
+
+        dag->getOutputs().swap( outputs);
+        auto expression = std::make_shared<ExpressionActions>(dag);
```
Comment (vdimir) [inline src/Storages/StorageValues.cpp:58]: Does this expression required for reordering columns in pipe output according to `column_names`?

**Review by vdimir — APPROVED** [review]

File: src/Storages/StorageValues.cpp (hunk 1 above)
Comment (KochetovNicolai) [reply src/Storages/StorageValues.cpp:58]: Yes, reordering and removing unneeded columns.

File: src/Storages/StorageBuffer.cpp [hunk 2]
```diff
@@ -363,14 +365,31 @@ void StorageBuffer::read(
...
-        interpreter.buildQueryPlan(buffers_plan);
+        if (local_context->getSettingsRef().allow_experimental_analyzer)
+        {
+            auto storage = std::make_shared<StorageValues>(
+                    getStorageID(),
+                    storage_snapshot->getAllColumnsDescription(),
+                    std::move(pipe_from_buffers),
+                    *getVirtualsPtr());
+
+            auto interpreter = InterpreterSelectQueryAnalyzer(
```
Comment (novikd) [inline src/Storages/StorageBuffer.cpp:379]: Why do we need to create a `StorageValues`?

Comment (vdimir) [reply src/Storages/StorageBuffer.cpp:379]: As I get we just need any storage that returns data from `pipe_from_buffers`

**Pull request merged**